go/api/command_options.go

```diff
@@ -69,3 +73,39 @@ const (
 	UnixSeconds      ExpiryType = "EXAT"    // expire the value after the Unix time specified by [api.Expiry.Count], in seconds
 	UnixMilliseconds ExpiryType = "PXAT"    // expire the value after the Unix time specified by [api.Expiry.Count], in milliseconds
 )
+
+// LPosOptions represents optional arguments for the [api.ListCommands.LPosWithOptions] and
+// [api.ListCommands.LPosCountWithOptions] commands.
+//
+// See [valkey.io]
+//
+// [valkey.io]: https://valkey.io/commands/lpos/
+type LPosOptions struct {
+	// Represents if the rank option is set.
+	IsRankSet bool
```

Comment:
Setting this field could be easily forgotten. This should be checked in runtime or in compile time. Is there a way to use a builder or nullable/optional fields or factory methods/contructors instead?

Reply:
Added builder functions to set these options which would handle setting these bools as well. Hence, reader won't have to set these explicitly.